src/openassetio-core/include/openassetio/trait/TraitBase.hpp

```diff
@@ -17,6 +17,53 @@ namespace trait {
 /**
  * Abstract CRTP base class for specification traits.
  *
+ * This non-virtual templated base class provides the common interface
+ * for a concrete trait view on a specification::Specification.
+ *
+ * A trait view provides a way to hide the underlying dictionary-like
+ * data access from hosts and managers.
```

Comment:
Restructured slightly in bc850855 and fixed the example.

Reply:
At the risk of dragging this out further, the example in the docs is rather boilerplatey, reproducing it here:
```c++
property::Int myValue = 123;  // Default if property not found.

MyTrait myTrait{specification};

if (myTrait.isValid()) {

  if (myTrait.getMyValue(&myValue) != TraitPropertyStatus::kFound) {

    std::cerr << "Warning: myValue not found\n";
  }
}
```
We could simplify to remove `isValid` and not raise an exception if the trait is unsupported (can be checked via `Specification::hasTrait` so the functionality exists elsewhere), and instead add another `TraitPropertyStatus::kUnsupportedTrait`, then the workflow simplifies to
```c++
property::Int myValue = 123;  // Default if property not found.

if (MyTrait{specification}.getMyValue(&myValue) != TraitPropertyStatus::kFound) {

  std::cerr << "Warning: myValue not found\n";
}
```

Reply:
I guess this relates to the comment re. allowing mutators to set properties and implicitly make the trait supported, rather than requiring specifications to be pre-constructed with the supported traits.

Reply:
Like the proposition, shall we do it in a follow up PR?

Reply:
> Restructured slightly in [bc85085](https://github.com/TheFoundryVisionmongers/OpenAssetIO/commit/bc8508552b00ee92ace36906d9a647aeabe0b012) and fixed the example.

This is def on the right track, and the example is great. What do you think about moving it above all the requirement for a derived class?

Reply:
Move up in bdb6c289 - seems to read well like that.